### PR TITLE
Add tests for WeakStable and enable WeakStable logic

### PR DESCRIPTION
**Pull request by gmarek**
@kubernetes/sig-testing @spxtr

### CODE REVIEW

File: mungegithub/mungers/submit-queue_test.go [hunk 1]
```diff
@@ -409,8 +416,10 @@ func TestSubmitQueue(t *testing.T) {
 		events           []github.IssueEvent
 		ciStatus         *github.CombinedStatus
 		jenkinsJob       jenkins.Job
-		lastBuildNumber  string
+		lastBuildNumber  int
```
Comment (spiffxp) [inline mungegithub/mungers/submit-queue_test.go]: just curious, what prompted the move from `string` to `int`?

Comment (gmarek) [inline mungegithub/mungers/submit-queue_test.go]: Necessity of computing `value - 1` for weakStable check.

File: mungegithub/mungers/submit-queue.go [hunk 2]
```diff
@@ -259,9 +259,8 @@ func (sq *SubmitQueue) AddFlags(cmd *cobra.Command, config *github.Config) {
 		"kubernetes-e2e-gce-scalability",
 		"kubernetes-kubemark-5-gce",
 	}, "Comma separated list of jobs in Jenkins to use for stability testing")
-	cmd.Flags().StringSliceVar(&sq.WeakStableJobNames, "weak-stable-jobs", []string{
-		"kubernetes-kubemark-500-gce",
```
Comment (eparis) [inline mungegithub/mungers/submit-queue.go]: This test shouldn't be here any more?

File: mungegithub/mungers/submit-queue_test.go [hunk 3]
```diff
@@ -164,13 +165,19 @@ func FailJenkins() jenkins.Job {
 	}
 }
 
+func getJUnit(testsNo int, failuresNo int) []byte {
```
Comment (eparis) [inline mungegithub/mungers/submit-queue_test.go]: Does this belong here or somehow over in the utils? Also does it matter that you are no longer hard coding `time=1234` ?

File: mungegithub/mungers/submit-queue.go (hunk 2 above)
Comment (gmarek) [inline mungegithub/mungers/submit-queue.go]: I don't want to add the code that enables it, together with adding the test in a single PR.

File: mungegithub/mungers/submit-queue_test.go (hunk 3 above)
Comment (gmarek) [inline mungegithub/mungers/submit-queue_test.go]: No - I just found it useful for debugging to have different timestamps.

File: mungegithub/mungers/submit-queue_test.go [hunk 4]
```diff
@@ -746,6 +853,42 @@ func TestSubmitQueue(t *testing.T) {
...
+				data, err := json.Marshal(test.weakResults[buildNumberCopy])
+				if err != nil {
+					t.Errorf("Unexpected error: %v", err)
+				}
+				w.Write(data)
+			})
+		}
+		for junitFile, xml := range test.gcsJunit {
+			path = fmt.Sprintf("/bar/%v/artifacts/%v", test.lastBuildNumber, junitFile)
+			// workaround go for loop semantics
```
Comment (eparis) [inline mungegithub/mungers/submit-queue_test.go]: Isn't xml already a copy? What is the semantic issue here? (just for me knowledge)

File: mungegithub/mungers/submit-queue_test.go (hunk 1 above)
Comment (spiffxp) [inline mungegithub/mungers/submit-queue_test.go]: thanks

File: mungegithub/mungers/submit-queue_test.go (hunk 4 above)
Comment (gmarek) [inline mungegithub/mungers/submit-queue_test.go]: It's not really a copy. It's one of mine biggest issue with go. When you iterate over something, i.e. write `for x := range bla` compiler allocates a space for a single copy of element of `bla` and reuses it in all iterations. This is OK, for 'normal' processing, but it's kind of painful when you write a go routine inside a for loop or, like here, create a function to be called later.

**Pull request merged**